integration/query_evalution_test.go

```diff
@@ -0,0 +1,166 @@
+package integration
+
+import (
+	"math"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo"
+)
+
+func TestEvalMod(t *testing.T) {
+	t.Parallel()
+	ctx, collection := setup(t)
+
+	_, err := collection.InsertMany(ctx, []any{
+		bson.D{{"_id", "float64_1"}, {"value", float64(113.01)}},
+		bson.D{{"_id", "float64_2"}, {"value", float64(114.99)}},
+		bson.D{{"_id", "float64_3"}, {"value", float64(115.5)}},
+		bson.D{{"_id", "int64_1"}, {"value", int64(141)}},
+		bson.D{{"_id", "int64_2"}, {"value", int64(151)}},
+		bson.D{{"_id", "int64_3"}, {"value", int64(161)}},
+		bson.D{{"_id", "int32_1"}, {"value", int32(177)}},
+		bson.D{{"_id", "int32_2"}, {"value", int32(178)}},
+		bson.D{{"_id", "int32_3"}, {"value", int32(179)}},
+		bson.D{{"_id", "nil"}, {"value", nil}},
+		bson.D{{"_id", "string"}, {"value", "12"}},
```

Comment:
May you please add tests where the value is a weird number: ±Infinity, NaN, `math.MaxFloat64`, `math.SmallestNonzeroFloat64`, `math.MinInt64`? I suspect there might be a problem with large float64 -> int64 conversion (#371)